libcloud/test/storage/test_s3.py

```diff
@@ -905,6 +907,29 @@ def test_upload_big_object_via_stream(self):
         self.assertEqual(obj.name, object_name)
         self.assertEqual(obj.size, CHUNK_SIZE * 3)
 
+    def test_upload_object_via_stream_guess_file_mime_type(self):
+        def dummy_content_type(name):
+            return 'application/zip', None
+
+        old_func = libcloud.utils.files.guess_file_mime_type
+        libcloud.utils.files.guess_file_mime_type = dummy_content_type
```

Comment:
Can you please use monkeypatch.setattr from pytest? It's more robust: for example, if the test fails before entering the `try` block, we will have modified `libcloud.utils.files.guess_file_mime_type` for the other tests too.

Reply:
Changed to use Python `mock.patch`.